librz/analysis/esil/esil_trace.c

```diff
@@ -193,6 +227,16 @@ static int trace_hook_mem_write(RzAnalysisEsil *esil, ut64 addr, const ut8 *buf,
 	return ret;
 }
 
+/**
+ * Get instruction trace from esil trace by index
+ * \param etrace RzAnalysisEsilTrace *, esil trace
+ * \param idx int, index of instruction
+ * \return RzILTraceInstruction *, instruction trace at index
+ */
+RZ_API RzILTraceInstruction *rz_analysis_esil_get_instruction_trace(RzAnalysisEsilTrace *etrace, int idx) {
```

Comment:
Add checks on `idx`. Is it within the bounds of the vector?

Reply:
Yes, or it will report an assertion fail from the vector when it exceed the bounds.